internal/services/vorpal.go

```diff
@@ -93,16 +93,22 @@ func executeScan(vorpalWrapper grpcs.VorpalWrapper, filePath string) (*grpcs.Sca
 	return vorpalWrapper.Scan(fileName, sourceCode)
 }
 
-func manageVorpalInstallation(vorpalParams VorpalScanParams, vorpalWrapper grpcs.VorpalWrapper) error {
+func manageVorpalInstallation(vorpalParams VorpalScanParams, vorpalWrapper grpcs.VorpalWrapper, vorpalWrappers VorpalWrappersParam) error {
 	vorpalInstalled, _ := osinstaller.FileExists(vorpalconfig.Params.ExecutableFilePath())
 
-	if vorpalParams.VorpalUpdateVersion || !vorpalInstalled {
-		if err := vorpalWrapper.HealthCheck(); err == nil {
-			_ = vorpalWrapper.ShutDown()
+	if !vorpalInstalled || vorpalParams.VorpalUpdateVersion {
+		if err := checkLicense(vorpalParams.IsDefaultAgent, vorpalWrappers); err != nil {
+			return err
 		}
-		if err := osinstaller.InstallOrUpgrade(&vorpalconfig.Params); err != nil {
+		newInstallation, err := osinstaller.InstallOrUpgrade(&vorpalconfig.Params)
+		if err != nil {
 			return err
 		}
+		if newInstallation && vorpalParams.VorpalUpdateVersion {
```

Comment:
```suggestion
		if newInstallation {
```
in a situation we check together that I removed the executable file- we also want to shut down the service